sdk/src/main/scala/io/horizen/account/mempool/MempoolMap.scala

```diff
@@ -275,22 +276,71 @@ class MempoolMap(
     txsList
   }
 
-  def mempoolTransactionsMap(executable: Boolean): TrieMap[SidechainTypes#SCP, TxByNonceMap] = {
-    val txsMap = TrieMap.empty[SidechainTypes#SCP, TxByNonceMap]
+  // method used by the txpool namespace rpc methods, it retrieve a map of executable or non-executable TxPoolTransaction
+  // ordered by address and nonce
+  def mempoolTransactionsMap(executable: Boolean): TrieMap[Address, mutable.SortedMap[BigInteger, TxPoolTransaction]] = {
+    val txsMap = TrieMap.empty[Address, mutable.SortedMap[BigInteger, TxPoolTransaction]]
     val mempoolIdsMap = if (executable)
       executableTxs
     else
       nonExecutableTxs
     for ((from, nonceIdsMap) <- mempoolIdsMap) {
-      val nonceTxsMap: mutable.TreeMap[BigInteger, SidechainTypes#SCAT] = new mutable.TreeMap[BigInteger, SidechainTypes#SCAT]()
-      for ((txNonce, txId) <- nonceIdsMap) {
-        nonceTxsMap.put(txNonce, getTransaction(txId).get)
-      }
-      txsMap.put(from, nonceTxsMap)
+      txsMap.put(from.asInstanceOf[AddressProposition].address(), retrieveTxPoolByNonceMap(nonceIdsMap))
+    }
+    txsMap
+  }
+
+  // method used by the txpool namespace rpc methods, it retrieve a map of executable or non-executable TxPoolTransaction
+  // ordered by nonce filtering by from address
+  def mempoolTransactionsMapFrom(executable: Boolean, fromAddress: Address): mutable.SortedMap[BigInteger, TxPoolTransaction] = {
+    val mempoolIdByNonceMapFrom = if (executable)
+      executableTxs.getOrElse(new AddressProposition(fromAddress), mutable.SortedMap.empty[BigInteger, ModifierId])
```

Comment:
Even if more readable unfortunately this implementation can't be used because the `nonEmpty` method can't be applied on the `mempoolIdByNonceMapFrom` because it is an Option of object and not the direct object

Reply:
@saratnt we can't relay on the fact that `executableTxs` or `nonExecutableTxs` have a record for `fromAddress`
So the suggested approach may throw an unexpected exception.
Am I right?

Reply:
@davehorizen Option actually has nonEmpty method, but if you prefer you can use `isDefined` instead, it is the same
@i-Alex  nonExecutableTxs.get(new AddressProposition(fromAddress)) returns an Option that, if notEmpty == true , using get won't throw an exception.
Maybe it is more clear this way:

```
    val mempoolIdByNonceMapFrom = if (executable)
      executableTxs.get(new AddressProposition(fromAddress))
    else
      nonExecutableTxs.get(new AddressProposition(fromAddress))

    mempoolIdByNonceMapFrom match {
      case Some(map) => retrieveTxPoolByNonceMap(map)
      case None => mutable.SortedMap.empty[BigInteger, TxPoolTransaction]
    }

```

Reply:
ok, clear.
@davehorizen let's do as Sara suggested.

Reply:
looks fine